### ISSUE TITLE
Prepare Release 3.8.2

### ISSUE THREAD

**Issue by ann0see** [jamulussoftware/jamulus#2247]
**Target timeline**
Scheduled feature freeze / Start of translation process: 2022-01-27
Targeted translation completion date: ~~2022-02-06~~ 2022-02-13
~~Approximate~~ release date: 2022-02-20
Current state: Released  

**Checklist**
- [ ] Assign this issue to the release shepherd who is in charge of managing this checklist.
- [x] Pin this issue
- [x] Ensure that all issues/PR targeted for this release are done by checking the Project board with [the appropriate filter for this release](https://github.com/orgs/jamulussoftware/projects/2?card_filter_query=milestone%3A%22release+3.8.2%22). Remind main developers to review entries in *Waiting on team* state. 
- [x] Agree to de-tag unfinished Issues/PRs.
- [x] Declare a freeze for code and website by updating this Issue and adding a comment. PRs can still be worked on and may get reviewed, but must not be merged unless agreed explicitly.
- [x] Check the `needs documentation` label for any outstanding PRs flagged for this release and remove that label if done.
- [x] Check ./Jamulus -h output against the Include-Client/Server-Commands.md pages (wiki/en) and man page (distributions/Jamulus.1). Update if necessary.
- [x] Write the Changelog based on the list of PRs in "done" state ([example for 3.8.1](https://github.com/orgs/jamulussoftware/projects/2?card_filter_query=milestone%3A%22release+3.8.1%22+is%3Amerged))
- [x] Start Website translations
  - [x] Check for broken links with a link checker on the `next-release` branch and fix them
  - [x] Open a Pull Request from `next-release` to release, set it as "Draft", sanity check for conflicts and any obvious problems. 
  - [x] Declare a full freeze of the `next-release` and `release` branch. No changes should be made from now on to ensure translators don't have to work twice.
  - [x] Check if the list of translators in `tools/create-translation-issues.sh`. Make sure issue text is up-to-date. Add any URLs that will need localisation into the "New/Changed screenshots" section.
  - [x] Create a translation issue for each language with `tools/create-translation-issues.sh` using `web` argument (see notes in script). 
  - [x] If anyone finds critical issues now, all translators must be made aware of them and all languages should be updated.
- [x] Start App translations
  - [x] Generate `.ts` files in master via `lupdate`
  - [x] Check if the list of translators in `tools/create-translation-issues.sh` is up-to-date
  - [x] Create a translation issue for each language with `tools/create-translation-issues.sh` using `app` argument.
- [x] [Tag a beta release](https://jamulus.io/contribute/Release-Process#steps-for-a-specific-release) 
  - [x] Inform emlynmac for signing on macOS, and upload signed binary from [his repo](https://github.com/emlynmac/jamulus/releases/) to [ours](https://github.com/jamulussoftware/jamulus/releases/)
  - [x] Announce the beta release on Github Discussions. Pin the thread.
  - [x] Get feedback on the stability and resource usage (memleaks?, crashes?, installation issues?) of the beta release
- [x] Finish App translations
  - [x] Review translation PRs according to [release process checklist](https://jamulus.io/contribute/Release-Process#3-update-the-ts-files-returned-by-translators)
  - [x] Wait for all PRs to be merged (missing translations will revert to English automatically).
  - [x] Check for conflicting accelerator keys (see `tools/checkkeys.pl`)
  - [x] Generate `.qm` files via `lrelease Jamulus.pro`  
- [x] Finish Website translations
  - [x] Wait for all PRs to be merged (missing translations will revert to English automatically)
  - [x] Check for broken links with a link checker locally (run `_po4a-tools/po4a-create-all-targets.sh` locally first)
- [x] Check the milestone for mergable stuff again
- [x] Update the changelog again (we did have post-beta PRs which do not have an entry yet, so this is indepdendent of the previous item)
- [x] [Tag a release candidate](https://jamulus.io/contribute/Release-Process#steps-for-a-specific-release) (inform emlynmac for signing on macOS and upload signed binary from his repo to ours).
  - [x] Announce the release candidate on Github Discussions. Pin the thread. Unpin and lock the beta thread.
  - [x] Draft an announcement, include all contributors via `tools/get_release_contributors.py`
- [x] [Update the version number in `Jamulus.pro` and add the release date to the Changelog header and commit](https://jamulus.io/contribute/Release-Process#steps-for-a-specific-release)
- [x] Tag this commit as `r3_y_z`
  - [x] Wait for the build to complete
  - [x] Contact emlynmac for signing on macOS and upload signed binary [from his](https://github.com/emlynmac/jamulus/releases/) repo [to ours](https://github.com/jamulussoftware/jamulus/releases).
  - [x] Do a smoke test for Windows/Mac/Linux -- Do the binaries start/connect properly? Can earlier Jamulus versions properly connect to a server based on the new release?
  - [x] [Force tag that tag as `latest` and push.](https://jamulus.io/contribute/Release-Process#if-this-is-a-proper-release-move-the-latest-tag)
  - [x] [Upload the artifacts to SourceForge and set defaults](https://jamulus.io/contribute/Release-Process#upload-binaries-to-sourceforge).
  - [x] Update download links on the website by editing `config.yml` in `next-release`
  - [x] Publish Website release by merging `next-release` into `release`
- [x] Announce the new release with a summary of changes (+ link to the changelog for details) and a link to the download page
    - [x] On Github Discussions in the Announcements section. Lock the announcement thread. Pin the thread. Unpin and lock release candidate thread.
    - [x] On Facebook in the group "Jamulus (official group)". Turn off replies.
- [x] Trigger the update notification by updating both Update Check Servers with the new version (@pljones for update02, email corrados for update01)
- [x] [Prepare `Jamulus.pro` (`dev` suffix) and ChangeLog (add a header) for the [next release](https://jamulus.io/contribute/Release-Process#make-the-master-branch-ready-for-post-release-development)
- [x] Check that all Issues and PRs tagged for this release are in Done/Closed state.
- [x] Close the release milestone in both jamulus and jamuluswebsite repos
- [x] Create a milestone for the next minor release in jamulus and jamuluswebsite repos
- [x] Update this template in https://jamulus.io/contribute/Release-Process with any improvements if needed.
- [x] Unpin and close this issue
- [ ] Determine if a release retrospective is needed, create on Discussions if required

**Comment by ann0see** [comment]
@jamulussoftware/maindevelopers opened a new release issue. 

What should be included? Based on https://github.com/jamulussoftware/jamulus/discussions/2246#discussioncomment-1975566

**Comment by ann0see** [comment]
I‘d declare this mostly as a bug fix release without many new features.

**Comment by softins** [comment]
The biggest question is: are there any things currently in `master` since 3.8.1 that we would _not_ want to include in 3.8.2?

**Comment by ann0see** [comment]
I doubt anything on master is critical.

**Comment by gilgongo** [comment]
> I doubt anything on master is critical.

Does that mean everything in that [list](https://github.com/jamulussoftware/jamulus/discussions/2246#discussioncomment-1975566) is going in then?

**Comment by pljones** [comment]
> Does that mean everything in that [list](https://github.com/jamulussoftware/jamulus/discussions/2246#discussioncomment-1975566) is going in then?

Plus some more since... :)  But they all look like they're "Mostly Harmless"...

**Comment by gilgongo** [comment]
OK we better tag the relevant stuff for the release then in that case so we can see what we're dealing with.

**Comment by softins** [comment]
I can't see any more open PRs that need to go into 3.8.2, nor issues that we need to address for it, so I suggest we declare a code freeze.

I can update the `.ts` files directly ready for the translators. There are more changes to them than one would expect for a bugfix release. Most of these are in fact not really new text, but reformatting of existing texts to use `%1`, `%2`, etc instead of chopping the strings up.

Who is going to be release shepherd? I'm happy to if no-one else wants to, but happy also to defer if someone does want to.

**Comment by softins** [comment]
I have written a ChangeLog for this release, and put it in #2249

**Comment by gilgongo** [comment]
OK I've opened The PR for the website https://github.com/jamulussoftware/jamuluswebsite/pull/686. Declaring documentation freeze now.

**Comment by hoffie** [comment]
As discussed via chat, translators have been pinged and [3.8.2beta1 has been tagged](https://github.com/jamulussoftware/jamulus/releases/tag/r3_8_2beta1).

@emlynmac Can you please sign the release artifact and reply here?

**Comment by hoffie** [comment]
@emlynmac has fetched the tag, I've taken the signed autobuild artifact and have replaced the unsigned one in our Github release as per the checklist.
@ann0see has posted the announcement, which is now pinned as well.

So... next steps will be translation PRs I guess.

**Comment by ann0see** [comment]
@jamulussoftware/translators Unfortunately there are new strings to translate for 3.8.2. #2356 introduced some translation-related changes. If you want to update them, please pull the repo, translate the missing strings and then submit a new Pull Request today or tomorrow.

**Comment by jujudusud** [comment]
> Pull Request today or tomorrow.
Possible for FR saturday or sunday depending when I will ne at home.

**Comment by hoffie** [comment]
So...
- @pljones and @softins made sure that translation stuff was correct.
- @softins tagged rc1.
- @emlynmac provided a signed Mac build.
- I've replaced the Mac build with @emlynmac's build.

**Comment by softins** [comment]
We need an announcement in Discussions. I could create on using 3.8.1rc1 as a template: https://github.com/jamulussoftware/jamulus/discussions/2062

**Comment by ann0see** [comment]
Chinese translation has been merged after rc1. Do we need another rc?

**Comment by pljones** [comment]
Let's see if anything else crops up this week and decide at the weekend.

**Comment by softins** [comment]
> Chinese translation has been merged after rc1. Do we need another rc? I wouldn’t…

Having seen the innocuousness of the changes, I don't think we need another rc.

**Comment by ann0see** [comment]
But we do need to run lrelease again?
